torch/onnx/_internal/exporter/_fx_passes.py

```diff
@@ -21,7 +21,10 @@ def decompose_with_registry(
     to_preserve = _decomp.get_preserve_ops()
     # We can only preserve implemented ops
     can_preserve = tuple(to_preserve.intersection(onnx_registered_ops))
-    return exported_program.run_decompositions(decomp_table, _preserve_ops=can_preserve)
+    for op in can_preserve:
```

Comment:
does onnx have some legacy code for specifying a preserve_list? Would it make sense to try to clean this up as part of this PR?

Reply:
Sorry what do you mean? Like whether this logic can be simplified?

Reply:
I just see a call to `_decomp.get_preserve_ops()` a few lines up, which seems kind of suspect give you are removing the "preserve_ops" API publicly. So do we want to kill that logic on the onnx side too?

Reply:
Yeah i will rewrite this tmrw

Reply:
Actually @justinchuby does this decomp table used in any other workflow than export? If not, this decomp table logic can be dramatically simplified.

Reply:
This is the only place. Feel free to improve, thanks!

Reply:
Just to confirm, you mean it is not used in export_legacy right? I think there is one flow that uses torchdynamo directly?